flyteidl/pyproject.toml

```diff
@@ -13,7 +13,7 @@ requires-python = ">=3.8,<3.13"
 dependencies = [
     'googleapis-common-protos',
     'protoc-gen-openapiv2',
-    'protobuf>=4.21.1,<5.0.0',
+    'protobuf>=4.21.1',
```

Comment:
I think it's fine to remove the upper bound. cc @eapolinario @wild-endeavor any concerns?